source/agora/consensus/data/Block.d

```diff
@@ -150,6 +150,34 @@ public struct Block
     public Hash[] merkle_tree;
 
 
+    /***************************************************************************
+
+        Update the Block header with new signature
```

Comment:
This should be:

```
Returns:
    a copy of this block with a different signature and validators bitmask
```

Because it's not updating the `this` reference.